File: frontend/console/src/api/signal-video.js
import Axios from "@/http";

export function createSignalVideo(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-video', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalVideo(signalVideoId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-video/${ signalVideoId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalVideo(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-video', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalVideo(signalVideoId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-video/${ signalVideoId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalVideoWithFlag(signalVideoId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-video/${ signalVideoId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalVideos() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-video/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalVideo (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-video/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal_video.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_VIDEO = "/config/signal-video"
const URL_ROUTER_SIGNAL_VIDEO_BY_ID = "/config/signal-video/:signal-video-id"
const URL_ROUTER_SIGNAL_VIDEO_QUERY = "/config/signal-video/query"
const URL_ROUTER_SIGNAL_VIDEO_ALL = "/config/signal-video/all"

func SignalVideoRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_VIDEO, config.CreateSignalVideo)
	r.GET(URL_ROUTER_SIGNAL_VIDEO_BY_ID, config.RetrieveSignalVideo)
	r.PUT(URL_ROUTER_SIGNAL_VIDEO, config.UpdateSignalVideo)
	r.DELETE(URL_ROUTER_SIGNAL_VIDEO_BY_ID, config.DeleteSignalVideo)
	r.POST(URL_ROUTER_SIGNAL_VIDEO_QUERY, config.QuerySignalVideo)
	r.GET(URL_ROUTER_SIGNAL_VIDEO_ALL, config.GetAllSignalVideos)
	return r
}
